### PR TITLE
Feature/comics list

### PR DESCRIPTION
**Pull request by MkNice**

### CODE REVIEW

**Comment by MkNice** [comment]
add comics-list and pagintaion and link in more info

File: src/app/components/heroes-list/heroes-list.component.ts [hunk 1]
```diff
@@ -20,11 +20,12 @@ export class HeroesListComponent implements OnInit, OnDestroy {
   public loading: boolean = true;
   public extraInfo: boolean = false;
   private destroy$: ReplaySubject<number> = new ReplaySubject<number>(1);
+  public linkCharacters: string = 'characters?';
```
Comment (Bogdan-N) [inline src/app/components/heroes-list/heroes-list.component.ts:24]: looks like useless variable

File: src/app/components/pagination/pagination.component.html [hunk 2]
```diff
@@ -5,4 +5,4 @@
   [maxSize]="5"
   [boundaryLinks]="true"
   (pageChange)="onPageChanged($event)"
-></ngb-pagination>
+></ngb-pagination>
```
Comment (Bogdan-N) [inline src/app/components/pagination/pagination.component.html]: ? empty line

File: src/app/components/pagination/pagination.component.ts [hunk 3]
```diff
@@ -18,30 +18,47 @@ export class PaginationComponent implements OnInit, OnDestroy {
   public collectionSize: number;
   private itemsPerPage: number = 5;
   private destroy$: ReplaySubject<number> = new ReplaySubject<number>(1);
+  @Input() link: string = '';
   @Output() nextHeroes = new EventEmitter<MarvelCharacters[]>();
+  @Output() nextComics = new EventEmitter<MarvelCharacters[]>();
```
Comment (Bogdan-N) [inline src/app/components/pagination/pagination.component.ts]: why do you use two the same inputs types?

File: src/app/components/pagination/pagination.component.ts [hunk 4]
```diff
@@ -18,30 +18,47 @@ export class PaginationComponent implements OnInit, OnDestroy {
...
 
+    this.apiService.getData(requestString)
+      .pipe(
+        tap((nextData: DataMarvel) => {
+          this.nextComics.emit(nextData.data.results);
+        }),
+        takeUntil(this.destroy$)
+      )
+      .subscribe();
+  }
```
Comment (Bogdan-N) [inline src/app/components/pagination/pagination.component.ts]: you don't need this

File: src/app/components/pagination/pagination.component.ts [hunk 5]
```diff
@@ -18,30 +18,47 @@ export class PaginationComponent implements OnInit, OnDestroy {
...
   onPageChanged(pageNumber) {
-    const requestString: string = `characters?offset=${(pageNumber * this.itemsPerPage) - 5}&limit=${this.itemsPerPage}`;
+
+    const requestString: string = `${this.link}offset=${(pageNumber * this.itemsPerPage) - 5}&limit=${this.itemsPerPage}`;
 
     this.apiService.getData(requestString)
       .pipe(
         tap((nextData: DataMarvel) => {
           this.nextHeroes.emit(nextData.data.results);
+          this.nextComics.emit(nextData.data.results);
```
Comment (Bogdan-N) [inline src/app/components/pagination/pagination.component.ts]: ???

File: src/app/components/search-result/search-result.component.ts [hunk 6]
```diff
@@ -12,10 +13,9 @@ export class SearchResultComponent implements OnInit {
   public searchString: string = '';
   public response: any;
 
-  constructor(private dataSearchService: DataSearchService,  public marvelService: MarvelService) { }
+  constructor(private dataSearchService: DataSearchService,  public marvelService: MarvelService, public apiService: APIService) { }
```
Comment (Bogdan-N) [inline src/app/components/search-result/search-result.component.ts]: in constructor you always should use 'private'

**Pull request merged**